### PR TITLE
feat(upload): Allow file description option for upload

### PR DESCRIPTION
**Pull request by danielkim328**

### CODE REVIEW

File: src/api/uploads/PlainUpload.js [hunk 1]
```diff
@@ -77,6 +77,7 @@ class PlainUpload extends BaseUpload {
         const attributes = JSON.stringify({
             name: this.fileName,
             parent: { id: this.folderId },
+            description: this.fileDescription,
```
Comment (j3tan) [inline src/api/uploads/PlainUpload.js]: Verify that this is not a breaking change and not dependent on a server api change?

Comment (danielkim328) [reply src/api/uploads/PlainUpload.js]: @j3tan I verified in EUA, where this param is undefined, our upload does not breaking with these changes. Is there anywhere else I should be testing?

Comment (j3tan) [reply src/api/uploads/PlainUpload.js]: sounds good, just wanted to make sure this was backwards compatible

File: src/api/uploads/BaseUpload.js [hunk 2]
```diff
@@ -49,6 +51,7 @@ class BaseUpload extends Base {
         const attributes = {
             name: this.fileName || name,
             parent: { id: this.folderId },
+            description: this.fileDescription,
```
Comment (jstoffan) [inline src/api/uploads/BaseUpload.js]: Does this need to be part of the preflight OPTIONS request? If so, why?

Comment (jstoffan) [reply src/api/uploads/BaseUpload.js]: @danielkim328, can you comment? It looks like this field is also sent in the `preflightSuccessHandler`. Why is it sent twice?

Comment (danielkim328) [reply src/api/uploads/BaseUpload.js]: Ideally we want to verify the file submission is valid in preflight check, which should include whether description CAN be applied, such as valid length, or allow description etc

Comment (jstoffan) [reply src/api/uploads/BaseUpload.js]: Okay. I'm not sure what "ideally" means in this case, but I'm not very familiar with this endpoint, so I'll take your word for it.

Comment (danielkim328) [reply src/api/uploads/BaseUpload.js]: Well, preFlightRequest is made BEFORE the actual upload occurs to check whether the upload WILL be accepted. As I understand, preFlightRequest currently check only for file name conflicts, but 'ideally' it should also check if fileDescription can also be applied.

**Pull request merged**